app/models/invitation_manager.rb

```diff
@@ -26,6 +26,15 @@ def send_course_emails(course)
   end
   handle_asynchronously :send_course_emails
 
+  def send_meeting_emails(meeting)
+    meeting.invitees.not_banned.each do |invitee|
+      invitation = MeetingInvitation.new(meeting: meeting, member: invitee, role: 'Participant')
+      MeetingInvitationMailer.invite(meeting, invitee, invitation).deliver_now if invitation.save
+    end
+    meeting.update_attribute(:invites_sent, true)
```

Comment:
We should not be able to ignore the `invites_sent` attribute  as it was put in place to prevent organisers from sending out multiple meeting invitations to members, so can we remove this line please. 

There is a little bit more work to be done to remove it entirely but as long as we don't set it to true it should be okay for now.

Reply:
Right, that makes sense. The reason I left it as it is was mostly because of this:
```rb
  # …
  # app/controllers/admin/meetings_controller.rb

  def invite
    if @meeting.invites_sent
      return redirect_to admin_meeting_path(@meeting),
                         notice: t('admin.messages.meeting.invitations_already_sent')
    end

    InvitationManager.new.send_meeting_emails(@meeting)

    redirect_to admin_meeting_path(@meeting), notice: t('admin.messages.meeting.sending_invitations')
  end

  # …
```

What would be the new behaviour once `invites_sent` is gone? Do we send duplicated invitations? Do we (add code to) skip people already invited? Do we don’t send anything since they have already been sent? And how do we identify that we have sent something, if needed? By the presence of at least one invite?

Sorry if there are too many questions to answer, but I would like to hear your take on this before I remove `invites_sent` — I’d be happy to. 😃

Reply:
#### TLDR; no extra code needs to be written as conditions to ensure that no duplicate MeetingInvitations are created and emailed out are already in place.

We don't send duplicate invitations. We [have a uniqueness validation]( https://github.com/codebar/planner/blob/master/app/models/meeting_invitation.rb#L8) on `MeetingInvitation` which will prevent the creation of more than one invitation for a member/meeting combination. This will cause line 32 on the code above to not save a new meeting invitation (nor send an email) and move on to the next meeting_invitation. It may not be optimal but it does the job.

I did briefly consider that we could maybe add a unique index of meeting_id and member_id on the meeting_invitation table, however, that will cause `.save` to throw an exception that we will have to handle (and I think more work than what is necessary to get this to work, given we use the same pattern for `WorkshopInvitation`).